python/codepipeline-docker-build/Base.py

```diff
@@ -0,0 +1,71 @@
+from aws_cdk import (
+    aws_s3 as aws_s3,
+    aws_ecr,
+    aws_codebuild,
+    aws_ssm,
+    core,
+)
+
+
+class Base(core.Stack):
+    def __init__(self, app: core.App, id: str, props, **kwargs) -> None:
+        super().__init__(app, id, **kwargs)
+
+        # pipeline requires versioned bucket
+        bucket = aws_s3.Bucket(
+            self, "SourceBucket",
+            bucket_name=f"{props.namespace.lower()}-{core.Aws.ACCOUNT_ID}",
+            versioned=True,
+            removal_policy=core.RemovalPolicy.DESTROY)
+        # ssm parameter to get bucket name laster
+        bucket_param = aws_ssm.StringParameter(
+            self, "ParameterB",
+            parameter_name=f"{props.namespace}-bucket",
+            string_value=bucket.bucket_name,
+            description='cdk pipeline bucket'
+        )
+        # ecr repo to push docker container into
+        ecr = aws_ecr.Repository(
+            self, "ECR",
+            repository_name=f"{props.namespace}",
+            removal_policy=core.RemovalPolicy.DESTROY
+        )
+
+        # codebuild project meant to run in pipeline
+        cb_docker_build = aws_codebuild.PipelineProject(
+            self, "DockerBuild",
+            project_name=f"{props.namespace}-Docker-Build",
+            build_spec=aws_codebuild.BuildSpec.from_source_filename(
+                filename='pipeline_delivery/docker_build_buildspec.yml'),
+            environment=aws_codebuild.BuildEnvironment(
+                privileged=True,
+
+            ),
+            # pass the ecr repo uri into the codebuild project so codebuild knows where to push
+            environment_variables={
+                'ecr': aws_codebuild.BuildEnvironmentVariable(
+                    value=ecr.repository_uri),
+                'tag': aws_codebuild.BuildEnvironmentVariable(
+                    value='cdk')
+            },
+            description='Pipeline for CodeBuild',
+            timeout=core.Duration.minutes(60),
+        )
+        # codebuild iam permissions to read write s3
+        bucket.grant_read_write(cb_docker_build)
+
+        # codebuild permissions to interact with ecr
+        ecr.grant_pull_push(cb_docker_build)
+
+        # update props to pass objects to another stack
+        props.bucket_name = bucket.bucket_name
+        props.bucket_arn = bucket.bucket_arn
+        props.bucket_obj = bucket
+        props.cb_docker_build = cb_docker_build
+        self.output_props = props
```

Comment:
I would skip modifying `props`, I feel like it's more complicated than necessary. Just save the updated props locally.

Reply:
I'm not following, what do you mean?